cloudpub/ms_azure/service.py

```diff
@@ -556,6 +566,31 @@ def get_modular_resources_to_publish(
         )[0]
         return [prod_res, plan_res, tech_config]
 
+    def compute_targets(self, product_id: str) -> List[str]:
+        """List all the possible publishing targets order to seek data from Azure.
+
+        It also returns the ordered list of targets with the following precedence:
+            ``preview`` -> ``live`` -> ``draft``
+
+        Args:
+            product_id (str)
+                The product_id to retrieve all existing submission targets.
+
+        Returns:
+            List[Str]: The ordered list with targets to lookup.
+        """
+        all_targets = ["preview", "live", "draft"]
```

Comment:
Why is `preview` before `live`?

Reply:
Because we need to first iterate over preview to detect faster if the SAS is already in preview. It was part of already existing logic which I opted to keep it.